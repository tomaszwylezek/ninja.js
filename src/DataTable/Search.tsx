import React, { FC } from "react";

export interface ISearchProps {
  onSearch: (event: any) => void;
}

const Search: FC<ISearchProps> = props => {
  const { onSearch } = props;

  return (
    <div className="p-b-1">
      <input
        type="search"
        className="form-control"
        placeholder="SÃ¸g brugere"
        onChange={onSearch}
      />
    </div>
  );
};

export default Search;
