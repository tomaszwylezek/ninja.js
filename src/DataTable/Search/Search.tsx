import React, { FC, SyntheticEvent } from 'react';

export interface ISearchProps {
  value: string;
  handleChange: (event: SyntheticEvent<HTMLInputElement>) => void;
}

export const Search: FC<ISearchProps> = ({ value, handleChange }) => (
  <div className="p-b-1">
    <input type="search" className="form-control" placeholder="SÃ¸g brugere" onChange={handleChange} value={value} />
  </div>
);
