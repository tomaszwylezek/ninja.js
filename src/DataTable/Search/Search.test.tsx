import React from 'react';
import { mount, shallow } from 'enzyme';
import { Search, ISearchProps } from './Search';

describe('Search', () => {
  let props: ISearchProps;

  beforeEach(() => {
    props = {
      value: '',
      handleChange: jest.fn()
    };
  });

  it('renders without crashing', () => {
    const component = shallow(<Search {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('calls a function when button clicked', () => {
    const component = mount(<Search {...props} />);

    component.find('input').simulate('change');

    expect(props.handleChange).toHaveBeenCalled();
  });
});
