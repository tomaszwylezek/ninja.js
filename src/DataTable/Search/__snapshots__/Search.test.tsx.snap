// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders without crashing 1`] = `
<div
  className="p-b-1"
>
  <input
    className="form-control"
    onChange={[MockFunction]}
    placeholder="SÃ¸g brugere"
    type="search"
    value=""
  />
</div>
`;
